### ISSUE TITLE
Metals using the wrong JVM (W11/WSL, sdkman)

### ISSUE THREAD

**Issue by robmwalsh** [scalameta/metals#4995]
### Describe the bug

metals wouldn't work at all for me today. Stuck on "Importing Build"

found this gem in the logs:

```
2023.02.20 11:06:37 INFO  Attempting to connect to the build server...
2023.02.20 11:06:37 INFO  Running BSP server [/home/rob/.sdkman/candidates/java/22.0.0.2.r17-grl/bin/java, -Xms100m, -Xmx100m, -classpath, /home/rob/.sdkman/candidates/sbt/1.6.2/bin/sbt-launch.jar, -Dsbt.script=/home/rob/.sdkman/candidates/sbt/current/bin/sbt, xsbt.boot.Boot, -bsp]
2023.02.20 11:06:37 ERROR Running process '/home/rob/.sdkman/candidates/java/22.0.0.2.r17-grl/bin/java -Xms100m -Xmx100m -classpath /home/rob/.sdkman/candidates/sbt/1.6.2/bin/sbt-launch.jar -Dsbt.script=/home/rob/.sdkman/candidates/sbt/current/bin/sbt xsbt.boot.Boot -bsp' failedjava.io.IOException: Cannot run program "/home/rob/.sdkman/candidates/java/22.0.0.2.r17-grl/bin/java" (in directory "/home/rob/code/lsp"): error=2, No such file or directory
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1143)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
```
the issue here is that 22.0.0.2.r17-grl is not my active java installation, and I haven't even used it for ages
```
rob@desktop:~/code/lsp$ which java
/home/rob/.sdkman/candidates/java/current/bin/java

rob@desktop:~/code/lsp$ java -version
openjdk version "19.0.2" 2023-01-17
OpenJDK Runtime Environment Temurin-19.0.2+7 (build 19.0.2+7)
OpenJDK 64-Bit Server VM Temurin-19.0.2+7 (build 19.0.2+7, mixed mode, sharing)

rob@desktop:~/code/lsp$ echo ${JAVA_HOME}
/home/rob/.sdkman/candidates/java/current
```
Metals is not detecting a project at all, even though SBT is working fine
<img width="199" alt="image" src="https://user-images.githubusercontent.com/30517573/219993668-53c5d9bd-edef-46a4-8f4e-70a86a684c44.png">

I've tried `killall java -9`, restarting WSL, restarting my computer, changing the active jdk installation, nothing helps!

I'm currently using VSCode as a glorified text editor featuring syntax highlighting and not much else!

### Expected behavior

Metals should at least sort of work!

### Operating system

Windows

### Editor/Extension

VS Code

### Version of Metals

v0.11.10

**Comment by robmwalsh** [comment]
swapping to Bloop for the build server has got things working (about as well as one can expect with Bloop at least), but the java version is was still wrong (should be using jdk 19)

```
2023.02.20 13:28:02 INFO  Attempting to connect to the build server...
2023.02.20 13:28:02 INFO  Bloop uses /home/rob/.sdkman/candidates/java/17.0.2-tem defined at /home/rob/.bloop/bloop.json
2023.02.20 13:28:02 INFO  Attempting to connect to the build server...
2023.02.20 13:28:02 INFO  Bloop uses /home/rob/.sdkman/candidates/java/17.0.2-tem defined at /home/rob/.bloop/bloop.json
```

deleted the bloop.json file (and a created_by_metals.lock file in the same folder) and finally got things working!

**Comment by tgodzik** [comment]
Thanks for reporting! I think this is the case that Metals dealiased the SDKMan Java version previously and once it got removed we never updated the version to a new one. Though, I think we check if a Java home exist and report that to the user, but I would need to check that.

**Comment by tgodzik** [comment]
btw. did you set `metals.javaHome` to `/home/rob/.sdkman/candidates/java/current` or was the java home detected automatically?

**Comment by robmwalsh** [comment]
I don't believe I ever set metals.javaHome

**Comment by robmwalsh** [comment]
adding `"metals.javaHome": "/home/rob/.sdkman/candidates/java/current",` to my settings.json file has resolved the issue, and I'm now able to use SBT as my build server. I suspect this may break things if I used metals on windows due to settings sync but that's not going to be a problem for me (more for others that stumble across this issue).

On that note, I recently used a laptop that I haven't used for ages - only to use SSH remote to my usual machine, but I was still running vscode... I'll check the settings there. Settings sync may well explain this issue. I'll check it out later today.

**Comment by tgodzik** [comment]
The recommendation here would using `/home/rob/.sdkman/candidates/java/current`. LEt's reopen if there is anything that we can do on the metals side

**Comment by FreddieSanchez** [comment]
I had a similar issue after upgrading java using sdkman.  I went from Java 11 to 17.

```
  - Metals Server version: 1.3.5
  - Metals Java: 17.0.12 from Amazon.com Inc. located at ~/.sdkman/candidates/java/17.0.12-amzn
```

`JAVA_HOME` also pointed at this directory. 

I noticed that metals started using this version of java, but the build server (sbt) was started using the system installed version java21-openjdk. This caused my builds to fail due to having previously been compiled with a different version of java. 

The "fix" was to specify the `javaHome` setting as described above.
